### PR TITLE
Refactoring of testing infrastructure

### PR DESCRIPTION
**Pull request by rafie**

### CODE REVIEW

File: src/fork_gc.c [hunk 1]
```diff
@@ -844,6 +844,9 @@ static int recvCardvals(ForkGC *fgc, arrayof(CardinalityValue) *tgt, size_t *len
   if (*tgt) {
     rm_free(*tgt);
   }
+  if (*tgt) {
+    rm_free(*tgt);
+  }
```
Comment (oshadmi) [inline src/fork_gc.c:849]: This file should not be in this PR, right?
(and this snippet is redundant)

File: tests/pytests/test_vecsim.py [hunk 2]
```diff
@@ -1551,7 +1551,9 @@ def test_rdb_memory_limit():
         env.assertTrue(conn.execute_command('CONFIG SET', 'maxmemory', '0'))
 
 
-def test_timeout_reached():
+def test_timeout_reached(env):
+    if CODE_COVERAGE:
+        env.skip()
```
Comment (oshadmi) [inline tests/pytests/test_vecsim.py:1556]: No decreasing coverage?

**Review by oshadmi — COMMENTED** [review]
Few comments after a quick peek

File: src/fork_gc.c (hunk 1 above)
Comment (rafie) [reply src/fork_gc.c:849]: Yes, this was already merged to master, I'll remove it.

File: tests/pytests/test_vecsim.py (hunk 2 above)
Comment (rafie) [reply tests/pytests/test_vecsim.py:1556]: There's an issue on VecSim to fix this. In the meantime, skipped in order for tests to pass.

**Review by oshadmi — DISMISSED** [review]

**Pull request merged**